src/Raven.Server/Documents/Handlers/Streaming/StreamingHandler.cs

```diff
@@ -78,6 +78,55 @@ public Task StreamDocsGet()
             return Task.CompletedTask;
         }
 
+        
+        [RavenAction("/databases/*/streams/timeseries", "GET", AuthorizationStatus.ValidUser)]
+        public async Task Stream()
+        {
+            var documentId = GetStringQueryString("docId");
+            var name = GetStringQueryString("name");
+            var fromStr = GetStringQueryString("from", required: false);
+            var toStr = GetStringQueryString("to", required: false);
+            var offset = GetTimeSpanQueryString("offset", required: false);
+
+            var from = string.IsNullOrEmpty(fromStr) 
+                ? DateTime.MinValue 
+                : TimeSeriesHandler.ParseDate(fromStr, name);
+
+            var to = string.IsNullOrEmpty(toStr)
+                ? DateTime.MaxValue
+                : TimeSeriesHandler.ParseDate(toStr, name);
+
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var stats = context.DocumentDatabase.DocumentsStorage.TimeSeriesStorage.Stats.GetStats(context, documentId, name);
+                if (stats == default)
+                {
+                    // non existing time series
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
```

Comment:
Should this endpoint return 404? Are we returning 404 from other endpoints here?

Reply:
you are correct.